src/test/java/org/mskcc/cbio/oncokb/web/rest/SlackControllerIT.java

```diff
@@ -0,0 +1,257 @@
+    private static final String REGISTRATION_ADDR = "registration@example.com";
+    private static final String CONTACT_ADDR = "contact@example.com";
+    private static final String TECH_DEV_ADDR = "dev@example.com";
+    private static final String DEFAULT_ADDR = "default@example.com";
+
+    @Autowired
+    private ApplicationProperties applicationProperties;
+
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    private MailService mailService;
+    @Autowired
+    private JHipsterProperties jHipsterProperties;
+    @Autowired
+    private MessageSource messageSource;
+    @Autowired
+    private SpringTemplateEngine templateEngine;
+    @Autowired
+    private UserMailsService userMailsService;
+
+    private SlackService slackService;
+    @Autowired
+    private EmailService emailService;
+
+    @Autowired
+    private UserMapper userMapper;
+
+    @Spy
+    private Slack slack;
+
+    @Captor
+    private ArgumentCaptor<String> urlCaptor;
+
+    @Captor
+    private ArgumentCaptor<Payload> payloadCaptor;
+
+    @Spy
+    private JavaMailSenderImpl javaMailSender;
+
+    @Captor
+    private ArgumentCaptor<MimeMessage> messageCaptor;
+
+    @Autowired
+    private TokenServiceImpl tokenService;
+
+    @Autowired
+    private UserDetailsRepository userDetailsRepository;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    private SlackController slackController;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        MockitoAnnotations.initMocks(this);
+        doReturn(null).when(slack).send(urlCaptor.capture(), payloadCaptor.capture());
+        doNothing().when(javaMailSender).send(messageCaptor.capture());
+
+        // Specify webhook address for testing
+        applicationProperties.setSlack(new SlackProperties());
+        applicationProperties.getSlack().setUserRegistrationWebhook(USER_REGISTRATION_WEBHOOK);
+
+        // specify application emails for testing
+        applicationProperties.setEmailAddresses(new EmailAddresses());
+        applicationProperties.getEmailAddresses().setLicenseAddress(LICENSE_ADDR);
+        applicationProperties.getEmailAddresses().setContactAddress(CONTACT_ADDR);
+        applicationProperties.getEmailAddresses().setRegistrationAddress(REGISTRATION_ADDR);
+        applicationProperties.getEmailAddresses().setTechDevAddress(TECH_DEV_ADDR);
+        jHipsterProperties.getMail().setFrom(DEFAULT_ADDR);
+
+        mailService = new MailService(jHipsterProperties, javaMailSender, messageSource, templateEngine, userMailsService, applicationProperties);
+        slackService = new SlackService(applicationProperties, mailService, emailService, userMailsService, userMapper, slack);
+        slackController = new SlackController(userService, userRepository, mailService, slackService, userMapper);
+    }
+
+    @Test
+    void testApproveUser() throws IOException, MessagingException {
```

Comment:
This test mocks an approval payload for a mock user. Tests email integration, Slack integration, User activation, User Token, and UserMails. This is the main thing to be reviewed. If this type of test is what you are looking for I can make it for all the other cases.